### PR TITLE
Limit issue body to 500 characters

### LINKED ISSUE #486: HTTP 400 in command hacktoberissues when the issue description is too long
## Description
Discord does not allow embeds with description greater than 2040 in length. When `.hacktoberissues` comes across an issue with long enough body, the command does not respond and fails with the traceback shown in "Actual Behaviour section"

_Edit: Yes, it is rare, but I've encountered it more than 2 times in my test environment_ 

## Steps to Reproduce
Repetitively using `.hacktoberissues` command until it comes across an issue with long enough body.

## Expected Behaviour 
The bot could either paginate the description or put a limit on how much text in the description is shown. If I were to choose, I'd go with the later.

## Actual Behaviour
Command throws an exception
```
10/06/20 19:11:17 - root DEBUG: Error Encountered: HTTPException - 400 Bad Request (error code: 50035): Invalid Form Body
In embed.description: Must be 2048 or fewer in length., Command: hacktoberissues, Author: Quanta#8608, Channel: console
10/06/20 19:11:17 - bot.exts.evergreen.error_handler ERROR: Unhandled command error: 400 Bad Request (error code: 50035): Invalid Form Body
In embed.description: Must be 2048 or fewer in length.
Traceback (most recent call last):
  File "/home/quanta/.local/share/virtualenvs/seasonalbot-bEK16TGM/lib/python3.8/site-packages/discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "/home/quanta/Desktop/contrib/seasonalbot/bot/exts/halloween/hacktober-issue-finder.py", line 50, in hacktoberissues
    await ctx.send(embed=embed)
  File "/home/quanta/.local/share/virtualenvs/seasonalbot-bEK16TGM/lib/python3.8/site-packages/discord/abc.py", line 890, in send
    data = await state.http.send_message(channel.id, content, tts=tts, embed=embed,
  File "/home/quanta/.local/share/virtualenvs/seasonalbot-bEK16TGM/lib/python3.8/site-packages/discord/http.py", line 245, in request
    raise HTTPException(r, data)
discord.errors.HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
In embed.description: Must be 2048 or fewer in length.
```

## Known Impacted Platforms

- [x] Web
- [x] Desktop
- [x] Android App
- [x] iOS App

## Possible Solutions
- Limiting the characters from the issue description shown in the embed.
- Paginating the embed

## Would you like to implement a fix?
- [x] I'd like to implement the bug fix
- [ ] Anyone can implement the bug fix

### PR DESCRIPTION
**Pull request by quanta-kt**
## Relevant Issues
Closes #486 

## Description
This PR puts a limit on how long the response sent by the `.hacktoberissues` is going to be by limiting the issue body up to it's first 500 characters. This avoids http 400 errors (see #486 ) when dealing with issues with very long body.

## Did you:
- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] If dependencies have been added or updated, run `pipenv lock`?
- [x] **Lint your code** (`pipenv run lint`)?
- [x] Set the PR to **allow edits from contributors**?

### CODE REVIEW

File: bot/exts/halloween/hacktober-issue-finder.py
Comment (Akarys42) [inline bot/exts/halloween/hacktober-issue-finder.py:106]: I’m being very picky, but would it be possible to use only 3 dots here? That’s what we usually use.
```suggestion
        embed.description = body[:500] + '...' if len(body) > 500 else body
```

**Review by Akarys42 — APPROVED** [review]
Looks good to me, thanks for the contribution!

File: bot/exts/halloween/hacktober-issue-finder.py
Comment (Den4200) [reply bot/exts/halloween/hacktober-issue-finder.py]: You should just use `textwrap.shorten` instead.

**Review by kwzrd — COMMENTED** [review]
What is the motivation for `500`? The issue discusses that the embed description can be up to `2048` characters in length.

**Comment by quanta-kt** [comment]
@kwzrd 500 is just a short number which I think is decent amount characters for embed descriptions. I agree this is opinionated, (large embeds just don't look good to me/I feel it's bad UX). So being pointed out, I'll limit it back to 2048 characters if that's agreeable.

**Comment by kwzrd** [comment]
It's not necessarily a bad choice, if you think that `500` looks and works better. I was just wondering about why it was chosen. Maybe you could show a screenshot of 500 vs 2048?

**Comment by quanta-kt** [comment]
@kwzrd Here are the screenshots you requested-
https://ibb.co/M21rqX2
https://ibb.co/gDM642P
The one with `2048` characters totally covers my laptop screen.

**Comment by kwzrd** [comment]
@quanta-kt Thanks for the examples, I agree that 500 is a more sane value, especially since you can just click the link to read more.

---

@Den4200 @Akarys42 Are you sure that `textwrap.shorten` is the better choice here? The problem with it is that it removes newlines. Consider [this issue](https://github.com/vJechsmayr/PythonAlgorithms/issues/382), which gets formatted like so:

image

If we just slice the string, we get to maintain the newlines, e.g. [this issue](https://github.com/BitByte-TPC/beginners-byte/issues/3):

image

**Comment by Den4200** [comment]
Ahh right, I forgot that `textwrap.shorten` removes newlines. Good catch,
sorry @quanta-kt.

**Comment by quanta-kt** [comment]
TextWrapper apparently has a [replace_whitespace](https://docs.python.org/3/library/textwrap.html#textwrap.TextWrapper.replace_whitespace) keyword-arg but `shorten` simply ignores it, weird.

**Review by kwzrd — APPROVED** [review]
Thanks for the contribution, it works great!

Before we merge this, we would be interested in cleaning up the history because there are 5 commits for a 1-line change. In practice, what we would want to do is only keep the first 2 commits, and drop the last 3. This is not your fault, as we told you to make a change and then asked you to revert it.

Are you familiar with `git rebase` and would you feel confident trying to do this? If not, that's ok, a core dev can help you or do it for you.

**Comment by quanta-kt** [comment]
Yes I am familiar with rebasing but still a little hesitant to do it, can you check this dummy PR [here](https://github.com/quanta-kt/seasonalbot/pull/2) and confirm if that's what I'm supposed to be doing?

Commands used-
```bash
git rebase -i issue-486~5 issue-486 # And picking only the first commit
git push origin +issue-486 # A forced push
```

Also, shouldn't it get squashed into a single commit automatically if one chooses to "Squash and merge" on GitHub?

**Comment by kwzrd** [comment]
Your dummy PR looks good, but you would want to keep the 2nd commit as well (reducing the dots from 4 to 3).

I think there is an easier way, we actually don't need to rebase because we only need to drop the last 3 commits:
```p
git reset --hard HEAD~3
```
You can use `git log` to verify that only commits 5f02c61 and 8645da6 remain. Then `git status` should say that your local branch is 3 commits behind the remote. Finally `git push --force` will remove the commits from the remote as well.

> Also, shouldn't it get squashed into a single commit automatically if one chooses to "Squash and merge" on GitHub?

Yeah, we had a chat internally and we wanted to give you a chance to do it as it's both a good learning opportunity and also makes sure that the actual commits that get merged are authored by you, rather than the person who squashed them.

**Comment by kwzrd** [comment]
Lovely, thank you!

**Pull request merged**
